KTOR-7823 add devcontainer

stokado opened the pull request:
**Subsystem**
Infrastructure / dev environment

**Motivation**
Provide a dev container for external contributors. [KTOR-7823](https://youtrack.jetbrains.com/issue/KTOR-7823) Create a dev container for Ktor

**Solution**
Add a Dockerfile and devcontainer configuration.

stokado commented:
I was able to build a JVM target using this configuration.
However, `./gradlew check` failed with strange errors (e.g. IOException: Broken pipe).
I am opening this PR to gather feedback.
![image](https://github.com/user-attachments/assets/d291b14d-b869-4d8d-8b62-5314b538783c)

e5l commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,7 @@
+{
+  build: {
```
Could you please add quotes for the JSON keys?

e5l commented:
Hey @stokado, thanks for the PR! Could you check one minor? I failed to open this in dev container using extension

e5l commented:
There is also a warning reported from Gradle:
```
ERROR: JAVA_HOME is set to an invalid directory: /usr/lib/jvm/java-21-openjdk-amd64

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
```

stokado commented:
> Hey @stokado, thanks for the PR! Could you check one minor? I failed to open this in dev container using extension

I checked a dev container in IntelliJ IDEA Ultimate and in GitHub Codespaces
What extension did you try?

e5l commented:
vscode dev containers

stokado commented:
Alright, I will check vscode extension later

stokado commented:
I've updated configuration and replaced developer user with a root user
Unfortunately I was not able to build project in vscode extension with non-root user
I hope it's fine :)

I've also added quotes around keys in JSON and updated configuration of JAVA_HOME env variable
There was also 1 problem, why could not build project inside vscode extension, I had to replace Windows line separator in `gradlew` with Unix

I was able to sync project in both IntelliJ and vscode

osipxd commented:
I've added android build tools installation in a separate PR: https://github.com/ktorio/ktor/pull/4513

e5l commented:
@stokado, @osipxd thank you for the nice work!
